src/main/java/org/team1540/robot2025/subsystems/vision/apriltag/AprilTagVisionIOSim.java

```diff
@@ -0,0 +1,33 @@
+package org.team1540.robot2025.subsystems.vision.apriltag;
+
+import static org.team1540.robot2025.subsystems.vision.apriltag.AprilTagVisionConstants.*;
+
+import edu.wpi.first.math.geometry.Transform3d;
+import org.photonvision.simulation.PhotonCameraSim;
+import org.photonvision.simulation.SimCameraProperties;
+import org.photonvision.simulation.VisionSystemSim;
+import org.team1540.robot2025.RobotState;
+
+public class AprilTagVisionIOSim extends AprilTagVisionIOPhoton {
+    private final VisionSystemSim visionSim;
+
+    public AprilTagVisionIOSim(String cameraName, Transform3d cameraTransform) {
+        super(cameraName, cameraTransform);
+
+        this.visionSim = new VisionSystemSim(cameraName);
+        visionSim.addAprilTags(APRIL_TAG_FIELD_LAYOUT);
+
+        SimCameraProperties properties = new SimCameraProperties();
+        properties.setCalibration(SIM_RES_WIDTH, SIM_RES_HEIGHT, SIM_DIAGONAL_FOV);
+        properties.setFPS(SIM_FPS);
+        properties.setAvgLatencyMs(SIM_AVG_LATENCY_MS);
+
+        PhotonCameraSim cameraSim = new PhotonCameraSim(camera, properties);
+        visionSim.addCamera(cameraSim, cameraTransform);
+    }
+
+    public void updateInputs(AprilTagVisionIOInputs inputs) {
+        visionSim.update(RobotState.getInstance().getEstimatedPose());
```

Comment:
Seed this with `SimState.getInstance().getSimulatedRobotPose()` instead of using the estimated pose, since it's asking for the actual pose in simulation, not what's coming from the pose estimator (it doesn't make sense to seed the sim vision system with data from the pose estimator that depends on data from the vision system)